packages/test/service-load-test/src/runner.ts

```diff
@@ -92,6 +94,51 @@ async function runnerProcess(
     }
 }
 
+function scheduleFaultInjection(
+    ds: FaultInjectionDocumentServiceFactory,
+    container: Container,
+    runConfig: IRunConfig) {
+    const schedule = ()=>{
+        const leaveTime = runConfig.testConfig.readWriteCycleMs * 5 * Math.random();
+        printStatus(runConfig.runId, `fault injection in ${(leaveTime / 60000).toString().substring(0,4)} min`);
+        setTimeout(() => {
+            if(container.connected && container.resolvedUrl !== undefined) {
+                const deltaConn =
+                    ds.documentServices.get(container.resolvedUrl)?.documentDeltaConnection;
+                if(deltaConn !== undefined) {
+                    switch(Math.floor(Math.random() * 5)) {
+                        // dispreferr errors
+                        case 0: {
+                            // 1 in 10 chance of non-retritable error
+                            const canRetry = Math.floor(Math.random() * 10) === 0 ? false : true;
+                            deltaConn.injectError(canRetry);
+                            printStatus(runConfig.runId, `error injected canRetry:${canRetry}`);
```

Comment:
I think these would be helpful to have in the Kusto logs

Reply:
i'd be worried it's too noisy. the error identify themselves as fault injection errors, which should make them easily identifiable in the logs

Reply:
The more i think about this the more i am against it. if we can't diagnose issues with the existing logs, then the logs are incomplete, we shouldn't depend on test logs to figure things out.

Reply:
I agree about not using them as a crutch. But we may wonder sometimes whether it was the fault injection or, say, a bug in the service that was causing "noise" in the delta connection, and it would be nice to be able to quickly "root cause" these down to "by design".